test/e2e/clusterclass_rollout.go

```diff
@@ -142,7 +148,7 @@ func ClusterClassRolloutSpec(ctx context.Context, inputGetter func() ClusterClas
 			WaitForMachineDeployments:    input.E2EConfig.GetIntervals(specName, "wait-worker-nodes"),
 			WaitForMachinePools:          input.E2EConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
 		}, clusterResources)
```

Comment:
Sorry just realized this later. Can we add this in l.120
```go
		if input.FilterMetadataBeforeValidation == nil {
			input.FilterMetadataBeforeValidation = func(obj client.Object) clusterv1.ObjectMeta {
				return clusterv1.ObjectMeta{Labels: obj.GetLabels(), Annotations: obj.GetAnnotations()}
			}
		}
```

Then we can stop handling the nil case below and just use something like this everywhere

```go
				machineMetadata := filterMetadataBeforeValidation(machine)
```

Reply:
Looks great! Fixed it.